src/libsystemd/sd-journal/compress.c

```diff
@@ -630,35 +629,35 @@ int compress_stream_lz4(int fdf, int fdt, uint64_t max_bytes) {
         if (LZ4F_isError(c))
                 return -ENOMEM;
 
-        if (fstat(fdf, &st) < 0)
-                return log_debug_errno(errno, "fstat() failed: %m");
-
         frame_size = LZ4F_compressBound(LZ4_BUFSIZE, &preferences);
-        size =  frame_size + 64*1024; /* add some space for header and trailer */
-        buf = malloc(size);
-        if (!buf)
+        out_allocsize = frame_size + 64*1024; /* add some space for header and trailer */
+        out_buff = malloc(out_allocsize);
+        if (!out_buff)
+                return -ENOMEM;
+
+        in_allocsize = frame_size; // TODO: what's the optimal size?
```

Comment:
Not a compression expert, so looking for suggestion here: what's the best way to size the input buffer? Should it be some % of the output one?

Reply:
nvm we compress at most LZ4_BUFSIZE anyway